web/clusters.go

```diff
@@ -292,7 +293,29 @@ func NewClusterListHandler(clustersService services.ClustersService) gin.Handler
 	return func(c *gin.Context) {
 		query := c.Request.URL.Query()
 
-		clusterList, err := clustersService.GetAll(query)
+		clustersFilter := &services.ClustersFilter{
```

Comment:
This way of binding looks quite anti gin-gonic. What about using `c.Bind` here?
Here an example: https://github.com/gin-gonic/gin/issues/742
In summary, we would need a struct (`ClustersFilter`) which would get all the information.
I would give a try at least

Reply:
@arbulu89 thanks i will try this out!

Reply:
@arbulu89 i tried this and i am not sure since it involves adding a ``form`` annotation to the `ClustersService` struct while I would prefer to keep the to layers decoupled or at least not having the service layer depending on the controller one at all.

Reply:
Well, I guess the ideal would be to have an additional struct that receives the query data with the `form` annotations, and it can be converted to the other page filter. Now that we have api structs, entities, models, yet another layer hehe

Anyway, feel free to implement as you wish. I just thought that using the gin gonic native way to deal with query parameters was better.

I will remove my request changes review.